webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js

```diff
@@ -6,31 +6,41 @@ import {
   AUTO_COMPLETE_RESET,
 } from './AutoCompleteConstants';
 
-const initialState = Immutable({
+const initialAutocompleteState = {
   controller: null,
   error: null,
   results: [],
   searchQuery: '',
   status: null,
   trigger: null,
-});
+};
 
-export default (state = initialState, action) => {
+export default (state = Immutable({}), action) => {
   const {
     type,
-    payload: { controller, error, results, searchQuery, status, trigger } = {},
+    payload: {
+      controller,
+      error,
+      results,
+      searchQuery,
+      status,
+      trigger,
+      id,
+    } = {},
   } = action;
   switch (type) {
     case AUTO_COMPLETE_REQUEST:
-      return state.merge({
+      return state.setIn([id], {
+        ...state[id],
```

Comment:
I don't see a reason doing `...state[id]` when you override all the fields later.

Reply:
the `results` is missing here,
and in my opinion it's a good convention to have for future changes